app/controllers/registration_controller.rb

```diff
@@ -0,0 +1,53 @@
+class RegistrationController < ApplicationController
+  def new
+    form_options
+  end
+
+  def create
+    form_options
+    @endpoint = registration_endpoint
+    @headers = registration_headers
+    @registration_args = registration_args
+    render 'show'
+  end
+
+  private
+
+  def form_options
+    @organizations = Organization.authorized(:view_organizations).select(:id, :name)
+    @locations = Location.authorized(:view_locations).select(:id, :name)
+    @host_groups = Hostgroup.authorized(:view_hostgroups).select(:id, :name)
+    @operating_systems = Operatingsystem.authorized(:view_operatingsystems).select(:id, :title)
+    @smart_proxies = SmartProxy.authorized(:view_smart_proxies).select(:id, :name)
```

Comment:
This should be only smart proxies with the Register feature (depends on https://github.com/theforeman/foreman/pull/8024). IMHO you shouldn't add Smart Proxy support in this page without that. My suggestion would be to leave out the Smart Proxy altogether until the Register module support is actually there.